Build error on todays code ( 4-19-23)

[  1%] Building CXX object src/celengine/CMakeFiles/celengine.dir/asterism.cpp.o
In file included from /usr/include/c++/7/memory:80:0,
                 from /DATA/SUSE/src/Celestia-master/src/celengine/asterism.cpp:10:
/usr/include/c++/7/bits/unique_ptr.h: In instantiation of ‘void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = StarNameDatabase]’:
/usr/include/c++/7/bits/unique_ptr.h:263:17:   required from ‘std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = StarNameDatabase; _Dp = std::default_delete<StarNameDatabase>]’
/DATA/SUSE/src/Celestia-master/src/celengine/stardb.h:101:56:   required from here
/usr/include/c++/7/bits/unique_ptr.h:76:22: error: invalid application of ‘sizeof’ to incomplete type ‘StarNameDatabase’
  static_assert(sizeof(_Tp)>0,
                      ^
make[2]: *** [src/celengine/CMakeFiles/celengine.dir/build.make:76: src/celengine/CMakeFiles/celengine.dir/asterism.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:979: src/celengine/CMakeFiles/celengine.dir/all] Error 2
make: *** [Makefile:136: all] Error 2


OpenSUSE 15.4 , KDE  qt5 ,  gcc version 7.5.0 (SUSE Linux)

--- a/src/celengine/stardb.cpp
+++ b/src/celengine/stardb.cpp
@@ -35,7 +35,6 @@
 #include "category.h"
 #include "meshmanager.h"
 #include "parser.h"
-#include "starname.h"
 #include "value.h"
 
 using namespace std::string_view_literals;

--- a/src/celengine/stardb.h
+++ b/src/celengine/stardb.h
@@ -27,6 +27,7 @@
 #include "astroobj.h"
 #include "hash.h"
 #include "staroctree.h"
+#include "starname.h"
 
 
 class StarNameDatabase;

